### PR TITLE
feat(site): improve bulk delete flow

### PR DESCRIPTION
**Pull request by aslilac**
Provides more details about what you're actually doing, with some summaries and requiring multiple clicks, rather than just typing "DELETE" without any further detail

https://github.com/coder/coder/assets/418348/d3e0fc54-c0c2-4adc-a762-0c62f0dc485f

### CODE REVIEW

**Comment by aslilac** [comment]
I'll fix the test on Monday, but other than that, this is ready!

File: site/src/pages/WorkspacesPage/BatchActions.tsx [hunk 1]
```diff
@@ -68,77 +73,257 @@ type BatchDeleteConfirmationProps = {
...
+      </Stack>
+    </Stack>
+  );
+};
+
+const PersonIcon: FC = () => {
+  // This size doesn't match the rest of the icons because MUI is just really
+  // inconsistent. We have to make it bigger than the rest, and pull things in
+  // on the sides to compensate.
+  return <PersonOutlinedIcon css={{ width: 18, height: 18, margin: -1 }} />;
```
Comment (BrunoQuaresma) [inline site/src/pages/WorkspacesPage/BatchActions.tsx:298]: I hate MUI not being consistent 😢

**Review by BrunoQuaresma — APPROVED** [review]

File: site/src/pages/WorkspacesPage/BatchActions.tsx [hunk 2]
```diff
@@ -68,77 +73,257 @@ type BatchDeleteConfirmationProps = {
...
+    ...new Set(
+      checkedWorkspaces.flatMap((workspace) =>
+        workspace.latest_build.resources.map(
+          (resource) => resource.icon || getIconPathResource(resource.type),
+        ),
+      ),
+    ),
+  ].map((url) => (
+    <img key={url} alt="" aria-hidden css={{ ...visuallyHidden }} src={url} />
+  ));
```
[the remark this answers is not in the document]
Comment (aslilac) [reply site/src/pages/WorkspacesPage/BatchActions.tsx:135]: we could add that to these to make sure it snags them, but the problem is that even if we put this on the images where they normally appear, they're not inserted in the DOM until the user is already supposed to be seeing them. the only way to make them load faster is to tell the browser about these images in advance.

File: site/src/pages/WorkspacesPage/BatchActions.tsx (hunk 1 above)
Comment (aslilac) [reply site/src/pages/WorkspacesPage/BatchActions.tsx:298]: but we have an escape plan, at least!

**Pull request merged**